src/Label/LabelOptionsResolver.php

```diff
@@ -68,23 +69,72 @@ public function getLabelOptions(Order $order): array
     public function getDeliveryOptions(Order $order): array
     {
         $deliveryOptions = OrderSettingsFactory::create($order)->getDeliveryOptions();
+        $packageType     = $this->getPackageType($order, $deliveryOptions);
         $shipmentOptions = $this->getShipmentOptions(
             $deliveryOptions,
             $order->getProducts(),
             $order->getIdCarrier()
         );
 
-        if (isset($shipmentOptions['insurance'])) {
-            $shipmentOptions['insurance'] = false === $shipmentOptions['insurance'] ? 0 : 500;
+        if (isset($shipmentOptions['insurance']) && $deliveryOptions) {
+            $shipmentOptions['insurance'] = true === $shipmentOptions['insurance']
+                ? $this->getInsurance($deliveryOptions, $packageType, $order)
+                : 0;
         }
 
         return [
             'shipmentOptions' => $shipmentOptions,
-            'package_type'    => $this->getPackageType($order, $deliveryOptions),
+            'package_type'    => $packageType,
             'package_format'  => $this->getPackageFormat($order, $deliveryOptions),
         ];
     }
 
+    /**
+     * @param \MyParcelNL\Sdk\src\Adapter\DeliveryOptions\AbstractDeliveryOptionsAdapter $deliveryOptions
+     * @param int                                                                        $packageType
+     * @param \Gett\MyparcelBE\Model\Core\Order                                          $order
+     *
+     * @return int the amount in euro for which the package should be insured
+     */
+    public function getInsurance(AbstractDeliveryOptionsAdapter $deliveryOptions, int $packageType, Order $order): int
+    {
+        $grandTotal  = $order->getTotalProductsWithTaxes();
+        $psCarrierId = $order->getIdCarrier();
+
+        try {
+            $fromPrice   = CarrierConfigurationProvider::get($psCarrierId, Constant::INSURANCE_CONFIGURATION_FROM_PRICE);
+            $maxAmount   = CarrierConfigurationProvider::get($psCarrierId, Constant::INSURANCE_CONFIGURATION_MAX_AMOUNT);
+            $consignment = ConsignmentFactory::createByCarrierId($deliveryOptions->getCarrierId());
+            $consignment->setPackageType($packageType);
+        } catch (\Throwable $e) {
+            return 0;
+        }
+
+        if ($grandTotal < $fromPrice || ! $consignment->canHaveShipmentOption('insurance')) {
```

Comment:
Er komt een paar keer 'insurance' voor, misschien kan je daarvoor AbstractConsignment::SHIPMENT_OPTION_INSURANCE voor gebruiken

Reply:
Je hebt gelijk, ik vind zelf 'insurance' duidelijker ook vanwege dat de regels anders erg lang worden en gaan afbreken, maar het is niet netjes. Ik pas het aan.